invoke-ai/InvokeAI
# feat: default processors for controlnet & t2i adapter

## What type of PR is this? (check all applicable)



## Have you discussed this change with the InvokeAI team?

## Description

- Make default settings in backend narrowed to individual model type. Main models and ControlNet/T2I Adapters have default settings so far, but they are totally different - main get VAE, CFG Scale, etc, while the control adapters only get preprocessor.
- Add model probe logic to attempt to autoselect a preprocessor for each control adapter type.
- Add tests for probe logic.
- Update UI to use control adapter `default_settings` to autoselect processor.
- Update MM tab's default settings - main models have a separate UI from the control adapters.

## QA Instructions, Screenshots, Recordings

Use a fresh db so the probe can do its thing when it adds models. Suggest `use_memory_db` and `skip_model_hash` options in `invokeai.yaml` for speed.

- Select a canny model in MM tab. You should see default processor is canny.
- Add a ControlNet and select a canny model, processor should default to canny.
- Back to MM tab, change the default to something else.
- Change the model away from and back to canny - processor should default to the new selection.

Try another control net and/or t2i adapter.



## Merge Plan

This PR can be merged when approved
